src/api/app/models/user.rb

```diff
@@ -63,6 +63,7 @@ class User < ApplicationRecord
   has_many :rss_feed_items, -> { order(created_at: :desc) }, class_name: 'Notification::RssFeedItem', as: :subscriber, dependent: :destroy
 
   scope :all_without_nobody, -> { where('login != ?', nobody_login) }
+  scope :autocomplete_login, ->(prefix) { where('login LIKE ?', "#{prefix}%") if prefix.present? }
```

Comment:
Mmmm, why do you check that the prefix is present?
And if this is the case and is not present? what would happen?

Reply:
if the scope its called without any parameter, you get an ArgumentError error
if you pass nil or ```''``` and you dont check the prefix, then the '%' will be passed to the query forming a query like ```SELECT `users`.`login` FROM `users` WHERE (login LIKE '%')``` and probably returning all users?

Reply:
I'm adding a test case for that. I think that is probably the issue with using scope here. Normally scope behaves like a class method returning either scoped or all. So more or less like:

```
def self.autocomplete(prefix)
   if prefix.present?
    where('login LIKE ?', "#{prefix}%")
  else
    all
  end
end
```

Reply:
But the scope shouldn't care too much about the prefix length... I would move this control either to the #autocomplete method or to the controller side.